play-services-core/src/main/kotlin/com/google/android/gms/credential/manager/PasswordManagerActivity.kt

```diff
@@ -0,0 +1,99 @@
+const val EXTRA_KEY_UTM_CONTENT = "utm_content"
+
+private const val TAG = "PasswordManagerActivity"
+
+private const val PSW_MANAGER_PATH = "https://passwords.google.com/"
+
+class PasswordManagerActivity : AppCompatActivity() {
+
+    private lateinit var webView: WebView
+
+    private val accountName: String?
+        get() = runCatching { intent?.getStringExtra(EXTRA_KEY_ACCOUNT_NAME) }.getOrNull()
+    private val utmSource: String?
+        get() = runCatching { intent?.getStringExtra(EXTRA_KEY_UTM_SOURCE) }.getOrNull()
+    private val utmMedium: String?
+        get() = runCatching { intent?.getStringExtra(EXTRA_KEY_UTM_MEDIUM) }.getOrNull()
+    private val utmCampaign: String?
+        get() = runCatching { intent?.getStringExtra(EXTRA_KEY_UTM_CAMPAIGN) }.getOrNull()
+    private val utmContent: String?
+        get() = runCatching { intent?.getStringExtra(EXTRA_KEY_UTM_CONTENT) }.getOrNull()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Log.d(TAG, "onCreate: start")
+
+        val accounts = AccountManager.get(this).getAccountsByType(AuthConstants.DEFAULT_ACCOUNT_TYPE)
+        val realAccountName = accounts.find { it.name.equals(accountName) }?.name
+
+        Log.d(TAG, "realAccountName: $realAccountName")
+
+        val layout = RelativeLayout(this)
+        layout.addView(ProgressBar(this).apply {
+            layoutParams = RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT).apply {
+                addRule(RelativeLayout.CENTER_HORIZONTAL)
+                addRule(RelativeLayout.CENTER_VERTICAL)
+            }
+            isIndeterminate = true
+        })
+        webView = WebView(this).apply {
+            layoutParams = RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT)
+            visibility = View.INVISIBLE
+        }
+        layout.addView(webView)
+        setContentView(layout)
+        WebViewHelper(this, webView).openWebView(constructPswPath(), realAccountName)
+    }
+
+    private fun constructPswPath(): String {
+        val psmPath = StringBuilder().apply {
+            if (!utmSource.isNullOrEmpty()) {
+                if (isEmpty()) append("?") else append("&")
+                append("$EXTRA_KEY_UTM_SOURCE=$utmSource")
+            }
+            if (!utmMedium.isNullOrEmpty()) {
+                if (isEmpty()) append("?") else append("&")
+                append("$EXTRA_KEY_UTM_MEDIUM=$utmMedium")
+            }
+            if (!utmCampaign.isNullOrEmpty()) {
+                if (isEmpty()) append("?") else append("&")
+                append("$EXTRA_KEY_UTM_CAMPAIGN=$utmCampaign")
+            }
+            if (!utmContent.isNullOrEmpty()) {
+                if (isEmpty()) append("?") else append("&")
+                append("$EXTRA_KEY_UTM_CONTENT=$utmContent")
+            }
```

Comment:
If we invoke the URL with different or no UTM vales, will it change the displayed information? If no, we can completely remove them, improving user privacy and reducing code complexity.

Reply:
In general these values are just used by Google Analytics inside a website but I don't know about this specific site.

Reply:
I tested it and removing these parameters had no effect